stdlib/email/headerregistry.pyi

```diff
@@ -53,15 +63,19 @@ class AddressHeader:
     @classmethod
     def parse(cls, value: str, kwds: dict[str, Any]) -> None: ...
 
-class UniqueAddressHeader(AddressHeader): ...
+class UniqueAddressHeader(AddressHeader):
+    max_count: ClassVar[Literal[1]]
 
 class SingleAddressHeader(AddressHeader):
     @property
     def address(self) -> Address: ...
 
-class UniqueSingleAddressHeader(SingleAddressHeader): ...
+class UniqueSingleAddressHeader(SingleAddressHeader):
+    max_count: ClassVar[Literal[1]]
 
 class MIMEVersionHeader:
+    max_count: ClassVar[int | None]
+    def init(self, *args: Any, **kw: Any) -> None: ...
```

Comment:
It seems that the header classes are typically used through a (dynamically created) subclass in a way where `super().init(...)` arguments go to `BaseHeader.init`. We can annotate these `init` methods accordingly. For example, this one could be:
```python3
    def init(self, name, *, parse_tree, defects, version, major, minor) -> None: ...
```
where `name`, `parse_tree` and `defects` are as in `BaseHeader`, and the rest are for `kwargs.pop`.

Reply:
How I hate "clever" code.